src/src/PluginStructs/P121_data_struct.h

```diff
@@ -0,0 +1,20 @@
+#ifndef PLUGINSTRUCTS_P121_DATA_STRUCT_H
+#define PLUGINSTRUCTS_P121_DATA_STRUCT_H
+
+#include "../../_Plugin_Helper.h"
+#ifdef USES_P121
+
+#include <Adafruit_Sensor.h>
+#include <Adafruit_HMC5883_U.h>
+
+struct P121_data_struct : public PluginTaskData_base
+{
+  bool begin();
+
+  Adafruit_HMC5883_Unified mag = Adafruit_HMC5883_Unified(12345);
```

Comment:
Why not `Adafruit_HMC5883_Unified mag(12345);` ?

Reply:
It was also used in an example. It is the sensor id. But the I2C Adress is fixed… in my understanding, you can‘t use two modules on the same ESP. So I used the id from the example

Reply:
You can use an I2C port extender, so that's not an issue.
The difference in what I suggested is that you use the constructor without the need to copy it.

If the sensor ID must be different among multiple tasks using the same hardware, then it should be configurable, or if it doesn't matter what the value is, you can also use the TaskIndex (but then you have to give it as a parameter to the constructor of P121_data_struct.

Reply:
Good idea. Will do